include/rocksdb/utilities/transaction_db.h

```diff
@@ -31,6 +31,49 @@ enum TxnDBWritePolicy {
 
 const uint32_t kInitialMaxDeadlocks = 5;
 
+class BaseLockMgr;
+
+// A lock manager handle
+// The workflow is as follows:
+//  * Use a factory method (like NewRangeLockManager()) to create a lock
+//    manager and get its handle.
+//  * A Handle for a particular kind of lock manager will have extra
+//    methods and parameters to control the lock manager
+//  * Pass the handle to RocksDB in TransactionDBOptions::lock_mgr_handle. It
+//    will be used to perform locking.
+class LockManagerHandle {
+ public:
+  // PessimisticTransactionDB will call this to get the Lock Manager it's going
+  // to use.
+  virtual BaseLockMgr* getLockManager() = 0;
+
+  virtual ~LockManagerHandle(){};
+};
+
+// A handle to control RangeLockMgr (Range-based lock manager) from outside
+// RocksDB
+class RangeLockMgrHandle : public LockManagerHandle {
+ public:
+  virtual int set_max_lock_memory(size_t max_lock_memory) = 0;
+
+  class Counters {
+   public:
+    uint64_t escalation_count;
+    uint64_t current_lock_memory;
```

Comment:
Are the concept of lock memory and escalation count are specific for the range lock implementation based on range tree?

Reply:
`current_lock_memory` is the total amount of memory currently used for lock data structures. It is applicable even to Point Lock Manager (although the Point Lock Manager doesn't report the number currently). 

for `escalation_count`... it may depend. One of the goals of the Range-based Lock Manager is that it can naturally support  Lock Escalation, so I guess the answer is more "yes" than "no".   One can imagine a range-based Lock Manager that doesn't support lock escalation but it won't be as useful.